app/assets/javascripts/templates/data_portal/modals/modal-download-all.jst.ejs

```diff
@@ -0,0 +1,13 @@
+<div class="c-modal-download-all">
+  <h2 class="title -dark">Download</h2>
+  <p class="description">Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minima cumque asperiores nam enim! Veniam reprehenderit possimus, iure labore explicabo ipsum ducimus eligendi</p>
+
+  <form action="#" onSubmit class="c-form js-form">
+    <input type="email" name="email" class="js-email" placeholder="E-mail address" required />
+    <div class="terms-container">
+      <input type="checkbox" name="terms-and-conditions" class="js-terms" /><label for="terms-and-conditions">I accept to the <a href="/terms-of-use" target="_blank" rel="noopener noreferrer">terms and conditions.
```

Comment:
You didn't check VoiceOver did you?

The `for` attribute of a label needs to contain the id of the input, not the name. Replace `name` by `id` and it works perfectly.